source/containers.tex

```diff
@@ -11236,20 +11260,29 @@
 \indexlibraryglobal{as_bytes}%
 \begin{itemdecl}
 template<class ElementType, size_t Extent>
-  span<const byte, Extent == dynamic_extent ? dynamic_extent : sizeof(ElementType) * Extent>
+  span<const byte, @\seebelow@>
     as_bytes(span<ElementType, Extent> s) noexcept;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
 \effects
-Equivalent to: \tcode{return \{reinterpret_cast<const byte*>(s.data()), s.size_bytes()\};}
+Equivalent to:
+\begin{codeblock}
+return span<const byte, @\seebelow@>{reinterpret_cast<const byte*>(s.data()), s.size_bytes()};
+\end{codeblock}
+
+\pnum
+\remarks The second template argument of the returned \tcode{span} type is:
+\begin{codeblock}
+Extent == dynamic_extent ? dynamic_extent : sizeof(ElementType) * Extent
+\end{codeblock}
```

Comment:
This feels awkward to me given that we have an actual mechanism in the language for "I don't want to write this messy return type here". Did you consider:
* Change the return type of `as_bytes` to `auto`, 
* "Let `U` be `span<const byte, Extent == dynamic_extent ? dynamic_extent : sizeof(ElementType) * Extent>`.", and
* `return U{/*...*/};`?
(Ditto `as_writable_bytes`.)

Reply:
I did consider introducing a type (or the extent) with "Let" but for some reason I didn't consider using `auto`. The form I used is what Tomasz (paper author) proposed in a pending-LWG issue.

I'll use your suggestion and push a new commit.

Reply:
After trying a few ways to say it, I think the clearest is:

> _Effects:_ Equivalent to:
````
auto p = reinterpret_cast<const byte*>(s.data());
auto n = s.size_bytes();
if constexpr (Extent == dynamic_extent)
  return span<const byte>{p, n};
else
  return span<const byte, sizeof(ElementType) * Extent>{p, n};
````

Reply:
Works for me.

Reply:
I think we could factor out some more commonality with:

> Let `E` be `dynamic_extent` if `Extent == dynamic_extent`, and `sizeof(ElementType) * Extent` otherwise.
>
> _Effects_: Equivalent to: `return span<const byte, E>{reinterpret_cast<const byte*>(s.data()), s.size_bytes()};`

but this is much better in any case - feel free to ignore if you have better things to do than twaddle with this bit of wording over and over.